Source/WebCore/platform/graphics/ImageBuffer.cpp

```diff
@@ -183,15 +183,15 @@ FloatRect ImageBuffer::clampedRect(const FloatRect& rect)
 static RefPtr<ImageBuffer> copyImageBuffer(Ref<ImageBuffer> source, PreserveResolution preserveResolution)
 {
     if (source->resolutionScale() == 1 || preserveResolution == PreserveResolution::Yes) {
-        if (source->hasOneRef())
+        if (source->hasOneStrongRef())
             return source;
     }
```

Comment:
I'm not thrilled about this because it could be racy—for example, if `source` has its refcount increased between the `hasOneStrongRef()` check and returning from the function—but I'm not sure it's problem in practice.

To be fair, since `source` was previously `ThreadSafeRefCounted<>`, this change basically has the same behavior as the previous code.

@shallawa Do you have any feedback here?  For example, if `source` has a refcount of 1, does that infer that it is not being used on other threads, too?

Reply:
Moved `WebCore::ImageBuffer` to another bug/patch.